mllib/src/main/scala/org/apache/spark/mllib/linalg/EigenValueDecomposition.scala

```diff
@@ -79,6 +79,11 @@ private[mllib] object EigenValueDecomposition {
     // Mode 1: A*x = lambda*x, A symmetric
     iparam(6) = 1
 
+    require(n * ncv.toLong <= Integer.MAX_VALUE, "Large n and/or k will exceed " +
```

Comment:
I don't mean to belabor this, but the new second error message implies that k alone is the problem, and that k exceeds 2^31-1, and those aren't necessarily true. Concretely I'm suggesting something simple like

```
require(n * ncv.toLong <= Integer.MAX_VALUE && ncv * (ncv.toLong + 8) <= Integer.MAX_VALUE, 
        s"k = $k and/or n = $n are too large to compute an eigendecomposition")
```